pkg/route/ingress/ingress.go

```diff
@@ -585,14 +608,21 @@ func routeMatchesIngress(
 	path *networkingv1beta1.HTTPIngressPath,
 	secretLister corelisters.SecretLister,
 	serviceLister corelisters.ServiceLister,
+	host string,
+	hostIsWildcard bool,
 ) bool {
-	match := route.Spec.Host == rule.Host &&
+	var wildcardpolicycheck bool
+	match := route.Spec.Host == host &&
 		route.Spec.Path == path.Path &&
 		route.Spec.To.Name == path.Backend.ServiceName &&
-		route.Spec.WildcardPolicy == routev1.WildcardPolicyNone &&
 		len(route.Spec.AlternateBackends) == 0 &&
 		reflect.DeepEqual(route.Annotations, ingress.Annotations)
-	if !match {
+	if hostIsWildcard {
+		wildcardpolicycheck = route.Spec.WildcardPolicy == routev1.WildcardPolicySubdomain
+	} else {
+		wildcardpolicycheck = route.Spec.WildcardPolicy == routev1.WildcardPolicyNone
+	}
+	if !match && !wildcardpolicycheck {
 		return false
 	}
```

Comment:
This can be simplified as follows:
```go
	wildcardPolicy := routev1.WildcardPolicyNone
	if hostIsWildcard {
		wildcardPolicy = routev1.WildcardPolicySubdomain
	}
	match := route.Spec.Host == host &&
		route.Spec.Path == path.Path &&
		route.Spec.To.Name == path.Backend.ServiceName &&
		len(route.Spec.AlternateBackends) == 0 &&
		route.Spec.WildcardPolicy == wildcardPolicy &&
		reflect.DeepEqual(route.Annotations, ingress.Annotations)
	if !match {
		return false
	}
```